miio/gateway.py

```diff
@@ -529,45 +581,302 @@ def set_light(self, color_name, brightness):
         brightness_and_color = brightness_and_color_to_int(
             brightness, color_map[color_name]
         )
-        return self.send("set_rgb", [brightness_and_color])
+        return self._gateway.send("set_rgb", [brightness_and_color])
 
 
-class SubDevice:
-    def __init__(self, gw, sid, type_, _, __, ___):
-        self.gw = gw
-        self.sid = sid
-        self.type = DeviceType(type_)
+class SubDevice(Device):
+    """
+    Base class for all subdevices of the gateway
+    these devices are connected through zigbee.
+    """
```

Comment:
Didn't we discuss this already, that subdevices must not derive from `Device` due to cli issues? Please refactor this only to take gateway and dev_info as parameters.

Reply:
sorry I forgot, done now